airflow/models/skipmixin.py

```diff
@@ -218,10 +226,16 @@ def skip_all_except(
             for branch_task_id in list(branch_task_id_set):
                 branch_task_id_set.update(dag.get_task(branch_task_id).get_flat_relative_ids(upstream=False))
 
-            skip_tasks = [t for t in downstream_tasks if t.task_id not in branch_task_id_set]
-            follow_task_ids = [t.task_id for t in downstream_tasks if t.task_id in branch_task_id_set]
+            skip_tasks = []
+            for t in downstream_tasks:
+                downstream_ti = dag_run.get_task_instance(  # type: ignore[union-attr]
```

Comment:
What is this ignoring and can it be fixed?

Reply:
The problem relates to the fact that dag_run is the union of two types and DagRun, DagRunPydantic, and DagRunPydantic doesn't have `get_task_instance`

Reply:
Add an `assert isinstance(dag_run, DagRun)` above. Search for *AIP-44* to find some related assert patterns that you can follow.

Reply:
Got it, fixed